### PR TITLE
created calendar widget

### PR DESCRIPTION
**Pull request by sylvr3**

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
@sylvr3 #35 resolves some of the non-issues the linter is erroring out on here.

File: view/template/patient.hbs [hunk 1]
```diff
@@ -1,3 +1,116 @@
...
+.highlight {
+    border: 1px solid red;
+}
+
+
+.highlight2 {
+    border: 1px solid green;
+}
+
+</style>
```
Comment (ChristianMurphy) [inline view/template/patient.hbs]: `doctype`, `html`, `head`, and `script`s are inherited from `view/layout/default.hbs`
Add the new scripts there, and any other meta data you need set.

File: view/template/patient.hbs [hunk 2]
```diff
@@ -62,7 +175,18 @@
                 </table>
             </div>
         </div>
-    </div>
+
+<body>
+    <div class="date-picker"></div>
```
Comment (ChristianMurphy) [inline view/template/patient.hbs]: `body` is also inherited from `view/layout/default.hbs`

**Comment by sylvr3** [comment]
@ChristianMurphy I see that you merged that PR already. Could this also be due to a merge conflict?

And is it possible to access JSON object properties in html, outside of an html table (I'm trying to do a {{#each surveys}} inside my calendar functions but it can't find the JSON object)?

**Comment by ChristianMurphy** [comment]
The lint warning do not look to be due to a merge conflict.

Yes it is possible to access the survey properties
``` js
var surveys = JSON.parse('{{{ surveys }}}');
```
will take the server side data and add it to the client side `var` surveys.

**Comment by ChristianMurphy** [comment]
Question, what is the goal of this widget?

If it is to select a date, `jquery ui` datepicker is perfect.
If it is to display scheduled and completed surveys on a calendar.
`jquery ui` date picker may not be as useful, something like [bootstrap calendar](https://github.com/Serhioromano/bootstrap-calendar) might be a better starting point.

**Comment by sylvr3** [comment]
The latter is the goal.
I guess I will have to look into this, then (and undo my previous work).

**Comment by ChristianMurphy** [comment]
sorry :disappointed:
But good news, it should make your life easier
[a single function taking in a json object is all thats needed.] (https://github.com/Serhioromano/bootstrap-calendar#feed-function) :smile:

**Comment by sylvr3** [comment]
Had to use the command "nvm install stable" in order to install bower successfully. However, gulp commands are no longer working. :disappointed: I am still using node v4.2.1 (npm v2.14.7).

**Comment by ChristianMurphy** [comment]
first tip.
``` sh
npm i -g gulpjs/gulp-cli#4.0
```
most likely gulp got overwritten

second tip
This libary can be installed without using bower :smile: 
``` html
<link rel="stylesheet" href="https://cdn.rawgit.com/Serhioromano/bootstrap-calendar/master/css/calendar.min.css">
<script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>
<script src="https://cdn.rawgit.com/Serhioromano/bootstrap-calendar/master/js/calendar.min.js"></script>
```

third tip
``` js
var calendar = $('#calendar').calendar({
     tmpl_path: 'https://cdn.rawgit.com/Serhioromano/bootstrap-calendar/master/tmpls/',
     events_source: [
            {
                "id": 293,
                "title": "Event 1",
                "url": "http://example.com",
                "class": "event-important",
                "start": 12039485678000, // Milliseconds
                "end": 1234576967000 // Milliseconds
            }
        ]});
```

**Comment by sylvr3** [comment]
I am thinking of using a FullCalendar, rather than Bootstrap Calendar, since I cannot get the calendar to appear despite including all the libraries and feeding the array. What are your thoughts on FullCalendar?
https://github.com/fullcalendar/fullcalendar

**Comment by ChristianMurphy** [comment]
Bootstrap Calendar, Full Calendar either would work.
I figured out how to get bootstrap calendar working in #36 
Full Calendar should be fine as well.
Just avoid using bower in setup, we want to avoid additional setup steps.

**Comment by ChristianMurphy** [comment]
Looking at Full Calendar it appears to be more actively maintained and has more customization than Bootstrap Calendar.
I would recommend going with Full Calendar, and using Bootstrap Calendar as a fall back only if need be.

**Comment by ChristianMurphy** [comment]
And CDN js offers online versions that can be used to avoid `Bower` :+1: 
http://fullcalendar.io/download/
https://cdnjs.com/libraries/moment.js/

**Comment by ChristianMurphy** [comment]
closing in favor of #37

**Pull request closed without merging**